src/docker.rs

```diff
@@ -0,0 +1,105 @@
+use anyhow::{Context, Result};
+use std::io::Write;
+use std::path::PathBuf;
+
+use crate::image::Architecture;
+
+impl From<Architecture> for &str {
+    fn from(arch: Architecture) -> &'static str {
+        match arch {
+            Architecture::Arm64 => "linux/arm64",
+        }
+    }
+}
+
+#[cfg(not(feature = "mock"))]
+mod inner {
+    use super::*;
+
+    use std::fs::File;
+    use std::os::fd::AsFd;
+    use std::process::{Command, Stdio};
+
+    const COMPRESSION_LEVEL: u32 = 9;
+
+    fn compress_image(source: File, destination: File) -> Result<File> {
+        let mut source = std::io::BufReader::new(source);
+
+        let destination = std::io::BufWriter::new(destination);
+
+        let stream_builder = xz2::stream::MtStreamBuilder::new()
+            .threads(num_cpus::get() as u32)
+            .preset(COMPRESSION_LEVEL)
+            .encoder()
+            .context("pull_docker_image: could not create compression stream")?;
+
+        let mut stream = xz2::write::XzEncoder::new_stream(destination, stream_builder);
+
+        std::io::copy(&mut source, &mut stream)
+            .context("compress_docker_image: could not compress docker file")?;
+
+        let writer = stream.finish()?;
+        Ok(writer.into_inner()?)
+    }
```

Comment:
isn't it possible to reuse https://github.com/omnect/omnect-cli/blob/main/src/file/compression.rs ?